### DISCUSSION TITLE
How to replace image by dark-mode version in dark mode?

### DISCUSSION THREAD

**Discussion by benruijl** [quarto-dev/quarto-cli#6666] [Q&A] [answered]
### Description

I have an image that has a light-mode and dark-mode version that I want to use on my Quarto page. On most website I can do:

```html
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://symbolica.io/resources/completion.png">
  <source media="(prefers-color-scheme: light)" srcset="https://symbolica.io/resources/completion_light.png">
  <img width="600" alt="A demo of Symbolica" src="https://symbolica.io/resources/completion.png">
</picture>
```

but Quarto has its own light mode/dark mode toggle that ignores the preferred colour scheme. How can I modify this code so that it sees that the `body` has `quarto-light` or `quarto-dark` set?

**Comment by mcanouil** [comment, marked as answer]
There are several discussions on this here and a blog post to show the principle: https://mickael.canouil.fr/posts/2023-05-30-quarto-light-dark/

Edit:

- https://github.com/quarto-dev/quarto-cli/discussions/5900
- https://github.com/quarto-dev/quarto-cli/discussions/5758
- https://github.com/quarto-dev/quarto-cli/discussions/2588
- https://github.com/quarto-dev/quarto-cli/discussions/5439
- https://github.com/quarto-dev/quarto-cli/discussions/3417

**Comment by benruijl** [comment]
Thanks, you javascript snippet works beautifully!
